src/overview/overview.md

```diff
@@ -0,0 +1,643 @@
+> Thank you to everyone who contributed to this overview!
+
+> **NOTES ON THE OVERALL CHAPTER STRUCTURE**
+>
+> We _embrace_ the query architecture and talk about areas _backwards_ compared
+> to traditional compiler texts. Since information dependence of the various
+> `rustc` areas form a Direct Acyclic Graph (DAG) at the query view (not acyclic
+> at the areas view), it makes sense to start at the root -- the final
+> executable -- and traverse the graph in a breadth-first traversal, all the way
+> until we get to name resolution. At that point, we can zoom out and talk about
+> the non-querified areas (in either reverse order like the querified areas, or
+> forwards order like traditional passes-based compilers, depending on which
+> order feels more natural).
```

Comment:
I'm not super convinced by this. I find it easier to follow the flow of information in the parse->codegen direction. That way I can know what information is available at every stage. Similarly, whether an area is querified or not feels like an implementation detail; it doesn't affect the general understanding of which information is computed where.

I do agree that it's important for contributors to understand the querified nature of rustc. I think the walkthrough chapter would be a good place to walk through the "real" execution order, starting from the cli invocation all the way to parsing the files.

Reply:
Yeah, I could shuffle the organization around the "usual" way without it being *too* disruptive. I'll see what people think w.r.t. ordering.

Reply:
I flipped the summary ordering to go from parse->codegen. Thanks for the feedback.